libparsec/crates/platform_storage/src/native/workspace.rs

```diff
@@ -665,21 +664,15 @@ async fn db_list_manifests(
     let rows = sqlx::query("SELECT blob FROM vlobs LIMIT ?1 OFFSET ?2")
         .bind(limit)
         .bind(offset)
-        .fetch_many(executor);
-
-    let res: anyhow::Result<Vec<RawEncryptedManifest>> = rows
-        .filter_map(|row_res| async {
-            match row_res {
-                // `fetch_many` returns a single "left" element for the query result (containing, among others,
-                // the number of changes), then (what we are actually interested into here) one "right"
-                // elements for each row in the result.
-                Ok(sqlx::Either::Left(_res)) => None,
-                Ok(sqlx::Either::Right(row)) => Some(row.try_get(0).map_err(anyhow::Error::from)),
-                Err(e) => Some(Err(e.into())),
-            }
-        })
-        .try_collect()
-        .await;
+        .fetch_all(executor);
+
+    let res: anyhow::Result<Vec<RawEncryptedManifest>> = match rows.await {
+        Ok(v) => v
+            .iter()
+            .map(|row| row.try_get(0).map_err(anyhow::Error::from))
+            .collect(),
+        Err(e) => Err(e.into()),
```

Comment:
**nitpick (if-minor):** I would have used the `map_ok` and `err_into` from [`TryFutureExt`](https://docs.rs/futures/latest/futures/future/trait.TryFutureExt.html)

Reply:
Not sure how to use those, can you add a code suggestion?

Reply:
```suggestion
    rows
      .await
      .map_ok(|v| v
            .iter()
            .map(|row| row.try_get(0).map_err(anyhow::Error::from))
            .collect::<Vec<_>>()
      ) 
      .err_into::<anyhow::Error>()
```

I would also merge that block with the `fetch_all()` above